flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkMatchers.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.eventtime;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+/**
+ * Matchers for {@link Watermark}.
+ */
+public class WatermarkMatchers {
+
+	/**
+	 * Creates a matcher that matches when the examined watermark has the given timestamp.
+	 */
+	public static Matcher<Watermark> watermark(long timestamp) {
+		return new WatermarkMatcher(timestamp);
```

Comment:
nit: I think in this case it would be easier with `FeatureMatcher`

```
	public static Matcher<Watermark> watermark(long timestamp) {
		return new FeatureMatcher<Watermark, Long>(
			equalTo(timestamp),
			"a watermark with value",
			"value of watermark"
		) {
			@Override
			protected Long featureValueOf(Watermark actual) {
				return actual.getTimestamp();
			}
		};
	}
```

Reply:
that is indeed much simpler 👌